kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/extensions/CoroutineLoggingTestExecutionFilter.kt

```diff
@@ -0,0 +1,81 @@
+package io.kotest.engine.test.extensions
+
+import io.kotest.common.ExperimentalKotest
+import io.kotest.core.config.configuration
+import io.kotest.core.listeners.Listener
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestContext
+import io.kotest.core.test.TestResult
+import io.kotest.engine.test.withCoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlin.coroutines.AbstractCoroutineContextElement
+import kotlin.coroutines.CoroutineContext
+
+@ExperimentalKotest
+interface LogListener : Listener {
```

Comment:
Going back and forth on whether I should instead attach these logs to `TestResult` 🤔

Reply:
I think I don't want to piggyback on `TestResult` cause I like the idea of having multiple loggers